AK/Optional.h

```diff
@@ -307,7 +320,27 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> : public Optiona
     }
 
 private:
-    alignas(T) u8 m_storage[sizeof(T)];
+    ALWAYS_INLINE constexpr void construct_null_if_necessairy(bool should_construct = is_constant_evaluated())
+    {
+        // OPTIMIZATION: Only construct the `m_null` member when we are constant-evaluating.
+        // Otherwise, this generates an unnecessairy zero-fill.
+#if defined(AK_COMPILER_GCC) || defined(HAS_ADDRESS_SANITIZER)
```

Comment:
if address sanitizer complains, does that mean that we *do* actually end up accessing uninitialized bytes?

Reply:
This might have been a problem because the copy or move constructor wasn't trivial. I can do a sanitized compile to check if it is still necessary.

Reply:
Tests pass locally without `#if defined(HAS_ADDRESS_SANITIZER)`, we'll see what CI has to say...

Reply:
Looks like `#if defined(HAS_ADDRESS_SANITIZER)` is no longer necessary!